R/kernel.r

```diff
@@ -300,20 +314,30 @@ initialize = function(connection_file) {
 
 run = function() {
     while (TRUE) {
+        debug("main loop: beginning")
         zmq.poll(
             c(sockets$hb, sockets$shell, sockets$control),
             rep(.pbd_env$ZMQ.PO$POLLIN, 3))
-        
-        if(bitwAnd(zmq.poll.get.revents(1), .pbd_env$ZMQ.PO$POLLIN))
```

Comment:
Rather than just reordering them, which could cause a similar issue if we ever call poll on a control message, I think we should use `next` ([equivalent](http://code.hammerpig.com/loops-continue.html) to `continue` in Python) to jump back to the polling call after handling any message. That way, we never try to resume handling a poll result after dispatching messages.